packages/client/utils/reactjiPickerRecent.ts

```diff
@@ -0,0 +1,40 @@
+const DEFAULT_ORDER = {
+  heart: 0,
+  tada: 0,
+  smile: 0,
+  rocket: 0,
+  fire: 0,
+  white_check_mark: 0,
+  confused: 0,
+  cry: 0,
+  x: 0
+}
+
+const RECENT_REACTJI_PICKER = 'recentReactjiPicker'
+
+const getRecent = () => {
+  const localStorageRecent = localStorage.getItem(RECENT_REACTJI_PICKER)
+  return localStorageRecent === null ? DEFAULT_ORDER : JSON.parse(localStorageRecent)
+}
+
+const updateUsage = (emojiId: string) => {
+  const frequentlyUsed = getRecent()
+  frequentlyUsed[emojiId] || (frequentlyUsed[emojiId] = 0)
+  frequentlyUsed[emojiId] += 1
+  localStorage.setItem(RECENT_REACTJI_PICKER, JSON.stringify(frequentlyUsed))
+}
+
+const getRecentArray = () => {
+  const frequentlyUsed = getRecent()
+
+  const sortedFrequentlyUsed: [string, number][] = []
+  for (const emoji in frequentlyUsed) sortedFrequentlyUsed.push([emoji, frequentlyUsed[emoji]])
```

Comment:
Since we want to have all emojis in the result, we need to iterate over the keys of `DEFAULT_ORDER` in order to not miss any.
For style it would be nicer to use `map` instead, so together this could be
```
const sortedFrequentlyUsed = Object.keys(DEFAULT_ORDER).map(
  (emoji) => [emoji, frequentlyUsed[emoji] ?? 0]
)
```
If there is no value stored, we set the count to 0.

Reply:
I agree for the map for better styling. However, instead of `DEFAULT_ORDER`, I need to iterate over `frequentlyUsed` keys. Because in `getRecent()` I am using `DEFAULT_ORDER` if there is no value in local storage. Then I am adding new emojis on top of `DEFAULT_ORDER` in local storage
```
const getRecent = () => {
   const localStorageRecent = localStorage.getItem(RECENT_REACTJI_PICKER)
   return localStorageRecent === null ? DEFAULT_ORDER : JSON.parse(localStorageRecent)
 }
```

Reply:
You are right, I missed that in `updateUsage` you initialize the local storage with the full map. I thought we only store the values which are > 0.